prov/sockets/src/sock_epoll.c

```diff
@@ -108,10 +112,14 @@ void sock_epoll_close(struct sock_epoll_set *set)
 
 int sock_epoll_create(struct sock_epoll_set *set, int size)
 {
+	int ret;
 	set->size = size;
 	set->used = 0;
 	set->pollfds = calloc(size, sizeof(struct pollfd));
-	return set->pollfds ? 0 : -1;
+	ret = set->pollfds ? 0 : -1;
+	if (ret < 0)
+		free(set->pollfds);
```

Comment:
I'm not sure what this code is trying to do.  We free set->pollfds only if we can't allocate it?  You should just be able to return !set->pollfds